Source/Meadow.Foundation.Core/Leds/RgbPwmLedExtensions.cs

```diff
@@ -1,156 +1,191 @@
-        /// <param name="offDuration"></param>
-        /// <param name="highBrightness"></param>
-        /// <param name="lowBrightness"></param>
-        public static void StartBlink(this RgbPwmLed led, Color color, TimeSpan onDuration, TimeSpan offDuration, float highBrightness = 1f, float lowBrightness = 0f)
+        /// <param name="led">The LED</param>
+        /// <param name="color">The LED color</param>
+        /// <param name="onDuration">The duration the LED stays on</param>
+        /// <param name="offDuration">The duration the LED stays off</param>
+        /// <param name="highBrightness">The maximum brightness of the animation</param>
+        /// <param name="lowBrightness">The minimum brightness of the animation</param>
+
+        public static async Task StartBlink(this RgbPwmLed led, Color color,
+            TimeSpan onDuration, TimeSpan offDuration,
+            float highBrightness = 1f, float lowBrightness = 0f)
         {
-            if (highBrightness > 1 || highBrightness <= 0)
-            {
-                throw new ArgumentOutOfRangeException(nameof(highBrightness), "onBrightness must be > 0 and <= 1");
-            }
-            if (lowBrightness >= 1 || lowBrightness < 0)
-            {
-                throw new ArgumentOutOfRangeException(nameof(lowBrightness), "lowBrightness must be >= 0 and < 1");
-            }
-            if (lowBrightness >= highBrightness)
-            {
-                throw new Exception("offBrightness must be less than onBrightness");
-            }
+            ValidateBrightness(highBrightness, lowBrightness);
+
+            await StopAnimation(led);
 
-            lock (_syncRoot)
+            led.SetColor(color);
+
+            await StartBlink(led, onDuration, offDuration, highBrightness, lowBrightness);
+        }
+
+        /// <summary>
+        /// Start the Blink animation which sets the brightness of the LED alternating between a low and high brightness setting, using the durations provided.
+        /// </summary>
+        /// <param name="led">The LED</param>
+        /// <param name="onDuration">The duration the LED stays on</param>
+        /// <param name="offDuration">The duration the LED stays off</param>
+        /// <param name="highBrightness">The maximum brightness of the animation</param>
+        /// <param name="lowBrightness">The minimum brightness of the animation</param>
+
+        public static async Task StartBlink(this RgbPwmLed led,
+            TimeSpan onDuration, TimeSpan offDuration,
+            float highBrightness = 1f, float lowBrightness = 0f)
+        {
+            ValidateBrightness(highBrightness, lowBrightness);
+
+            await StopAnimation(led);
+
+            lock (syncRoot)
             {
-                led.Stop();
+                cancellationTokenSource = new CancellationTokenSource();
 
-                var animationTask = new Thread((s) =>
+                animationTask = new Task(() =>
```

Comment:
What if I want to have > 1 blinking LED?  You need to track all animation Tasks, not just the most recent, yes?